pkg/docker/config/testdata/helper.json

```diff
@@ -0,0 +1,6 @@
+{
+    "auths": {
+        "foobar.example.org": {}
```

Comment:
Hum, do we actually want commit to a design in which `credsStore` takes precedence over explicit entries for registries? (Note to self: What does Docker do?)

Reply:
This is the (sanitised) docker config from my macbook, it seems to put a dummy entry in the auths and rely on the credstore to do it.

```json
{
  "auths" : {
    "private.registry.yoda" : {

    },
  },
  "HttpHeaders" : {
    "User-Agent" : "Docker-Client/19.03.2 (darwin)"
  },
  "stackOrchestrator" : "swarm",
  "credsStore" : "desktop"
}
```

looking at the source for docker/cli it looks like it uses the default `credStore` before falling back to the `auths` list.
https://github.com/docker/cli/blob/master/cli/config/configfile/file.go#L290

But I did have some trouble following the docker/cli code.

Reply:
I changed the order of importance so it is more logical.